src/NHibernate.Test/Ado/GenericBatchingBatcherFixture.cs

```diff
@@ -102,6 +104,63 @@ public void MassivePerformanceTest(bool batched)
 			}
 		}
 
+		[Test]
+		public void InterceptorOnPrepareStatementTest()
+		{
+			var interceptor = new DatabaseInterceptor();
+			using (var sqlLog = new SqlLogSpy())
+			using (var s = Sfi.WithOptions().Interceptor(interceptor).OpenSession())
+			using (var tx = s.BeginTransaction())
+			{
+				s.SetBatchSize(5);
+				for (var i = 0; i < 20; i++)
+				{
+					s.Save(new VerySimple { Id = 1 + i, Name = $"Fabio{i}", Weight = 1.45 + i });
+				}
+
+				tx.Commit();
+
+				// Called for each insert statement and once for generating AbstractBatcher.CurrentCommand
+				Assert.That(interceptor.TotalCalls, Is.EqualTo(21));
```

Comment:
I tried to remove the call for `AbstractBatcher.CurrentCommand`, but I didn't found a clean solution to do it. The issue is that without calling `OnPrepareStatement` for the `AbstractBatcher.CurrentCommand` the log message will not contain the changes made by the method.

Reply:
Are you saying that now `DbCommand` is still generated for each unique statement ( I didn't realize it). So now interceptor call is duplicated for unique commands.

Also it seems old batcher would optimize this to single interceptor call. And with new batcher it's 21 intercpetor calls. Not nice...

Reply:
> Are you saying that now DbCommand is still generated for each unique statement

No, only two commands are generated. One is created for `AbstractBatcher.CurrentCommand`, only because it is needed for `AbstractBatcher` in order to work and the other command that is actually executed.

> Also it seems old batcher would optimize this to single interceptor call

You are right, I've modified the code to have only a single interceptor call.